src/hotspot/share/services/threadStackTracker.hpp

```diff
@@ -33,48 +33,24 @@
 #include "utilities/nativeCallStack.hpp"
 #include "utilities/linkedlist.hpp"
 
-class SimpleThreadStackSite;
-
-class SimpleThreadStack {
-  friend class SimpleThreadStackSite;
-private:
-  address _base;
-  size_t  _size;
-public:
-  SimpleThreadStack() : _base(NULL), _size(0) { }
-  bool equals(const SimpleThreadStack& s) const {
-    return base() == s.base();
-  }
-
-  size_t  size() const { return _size; }
-  address base() const { return _base; }
-private:
-  void set_size(size_t size)  { _size = size; }
-  void set_base(address base) { _base = base; }
-};
-
-class SimpleThreadStackSite : public AllocationSite<SimpleThreadStack> {
+class SimpleThreadStackSite : public AllocationSite {
+  const address _base;
+  const size_t  _size;
 public:
   SimpleThreadStackSite(address base, size_t size, const NativeCallStack& stack) :
-    AllocationSite<SimpleThreadStack>(stack, mtThreadStack) {
-    data()->set_size(size);
-    data()->set_base(base);
-  }
-
-  SimpleThreadStackSite(address base, size_t size) :
-    AllocationSite<SimpleThreadStack>(NativeCallStack::empty_stack(), mtThreadStack) {
-    data()->set_base(base);
-    data()->set_size(size);
-  }
+    AllocationSite(stack, mtThreadStack),
+    _base(base),
+    _size(size)
+  {}
```

Comment:
nit: can you put {} on the same line as _size ?

Reply:
Done.